amaranth/hdl/_ir.py

```diff
@@ -552,6 +552,8 @@ def _assign_port_names(self):
         assigned_names = {name for name, conn, dir in self.ports if name is not None}
         for name, conn, dir in self.ports:
             if name is None:
+                if conn.name == "": # Private Signal has no name so there's nothing to name this port.
+                    raise TypeError("Top-level ports with private Signals must be named")
```

Comment:
```suggestion
                    raise TypeError("Signals used as top-level ports must have a non-empty name")
```

Reply:
This message is a stronger requirement than the code actually enforces, but probably easier for users to understand. Should the code be changed too?

Reply:
I changed the message but not the code.

Reply:
Oh, I actually misread the code.